Mission3/src/routes/auth.js

```diff
@@ -0,0 +1,150 @@
+import { Router } from "express";
+import { PrismaClient } from "@prisma/client";
+import bcrypt from "bcrypt";
+import jwt from "jsonwebtoken";
+import authMiddleware from "../middlewares/auth.middleware.js";
+import {
+  validateSignUp,
+  validateLogin,
+  validateToken,
+} from "../middlewares/validator/validate.auth.js";
+
+const prisma = new PrismaClient();
+const router = Router();
+
+const createAccessToken = (userId) => {
+  return jwt.sign({ userId }, process.env.ACCESS_TOKEN_SECRET_KEY, {
+    expiresIn: "12h",
+  });
+};
+
+const createRefreshToken = (userId) => {
+  return jwt.sign({ userId }, process.env.REFRESH_TOKEN_SECRET_KEY, {
+    expiresIn: "7d",
+  });
+};
+
+router.post("/signup", validateSignUp, async (req, res, next) => {
+  try {
+    const { email, nickname, password } = req.body;
+
+    const existingUserByEmail = await prisma.user.findUnique({
+      where: { email },
+    });
+    if (existingUserByEmail) {
+      return res.status(409).json({ error: "이미 사용중인 이메일입니다." });
+    }
+
+    const existingUserByNickname = await prisma.user.findUnique({
+      where: { nickname },
+    });
+    if (existingUserByNickname) {
+      return res.status(409).json({ error: "이미 사용중인 닉네임입니다." });
+    }
+
+    const hashedPassword = await bcrypt.hash(password, 10);
+
+    const user = await prisma.user.create({
+      data: {
+        email,
+        nickname,
+        password: hashedPassword,
+      },
+    });
+
+    const userWithoutPassword = {
+      id: user.id,
+      email: user.email,
+      nickname: user.nickname,
+      createdAt: user.createdAt,
+      updatedAt: user.updatedAt,
+    };
```

Comment:
```suggestion
    const { password, ...rest } = user
```
위에처럼 하게 되면 rest 변수에 기존에 선언한 userWithoutPassword과 같은 구조를 만들 수 있습니다. 참고하세요